### ISSUE TITLE
Unable to install Memcached on Ubuntu PHP 5.6

### ISSUE THREAD

**Issue by elliotwms** [puphpet/puphpet#2162]
It appears to be related to the `phpapi-20121212` package

Here is the relevant extract from the provisioning script output

```
==> local: Info: Concat[10-default_vhost_443.conf]: Scheduling refresh of Class[Apache::Service]
==> local: Notice: /Stage[main]/Puphpet_apache/Apache::Vhost[default_vhost_443]/File[10-default_vhost_443.conf symlink]/ensure: created
==> local: Info: /Stage[main]/Puphpet_apache/Apache::Vhost[default_vhost_443]/File[10-default_vhost_443.conf symlink]: Scheduling refresh of Class[Apache::Service]
==> local: Notice: /Stage[main]/Puphpet_mysql/Puphpet::Php::Module[mysql]/Php::Module[mysql]/Package[PhpModule_mysql]/ensure: created
==> local: Info: /Stage[main]/Puphpet_mysql/Puphpet::Php::Module[mysql]/Php::Module[mysql]/Package[PhpModule_mysql]: Scheduling refresh of Service[php5.6-fpm]
==> local: Notice: /Stage[main]/Puphpet_php/Puphpet::Php::Module[intl]/Php::Module[intl]/Package[PhpModule_intl]/ensure: created
==> local: Info: /Stage[main]/Puphpet_php/Puphpet::Php::Module[intl]/Php::Module[intl]/Package[PhpModule_intl]: Scheduling refresh of Service[php5.6-fpm]
==> local: Info: Puphpet::Php::Module[intl]: Scheduling refresh of Service[php5.6-fpm]
==> local: Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install php5.6-memcache' returned 100: Reading package lists...
==> local: Building dependency tree...
==> local: Reading state information...
==> local: E: Unable to locate package php5.6-memcache
==> local: E: Couldn't find any package by regex 'php5.6-memcache'
==> local: Error: /Stage[main]/Puphpet_php/Puphpet::Php::Module[memcache]/Php::Module[memcache]/Package[PhpModule_memcache]/ensure: change from purged to present failed: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install php5.6-memcache' returned 100: Reading package lists...
==> local: Building dependency tree...
==> local: Reading state information...
==> local: E: Unable to locate package php5.6-memcache
==> local: E: Couldn't find any package by regex 'php5.6-memcache'
==> local: Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install php5-memcached' returned 100: Reading package lists...
==> local: Building dependency tree...
==> local: Reading state information...
==> local: Some packages could not be installed. This may mean that you have
==> local: requested an impossible situation or if you are using the unstable
==> local: distribution that some required packages have not yet been created
==> local: or been moved out of Incoming.
==> local: The following information may help to resolve the situation:
==> local: 
==> local: The following packages have unmet dependencies:
==> local:  php5-memcached : Depends: php5-common (>= 5.2.0) but it is not going to be installed
==> local:                   Depends: php5-common (< 6.0.0) but it is not going to be installed
==> local:                   Depends: phpapi-20121212
==> local: E: Unable to correct problems, you have held broken packages.
==> local: Error: /Stage[main]/Puphpet_php/Puphpet::Php::Module[memcached]/Php::Module[php5-memcached]/Package[PhpModule_php5-memcached]/ensure: change from purged to present failed: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install php5-memcached' returned 100: Reading package lists...
==> local: Building dependency tree...
==> local: Reading state information...
==> local: Some packages could not be installed. This may mean that you have
==> local: requested an impossible situation or if you are using the unstable
==> local: distribution that some required packages have not yet been created
==> local: or been moved out of Incoming.
==> local: The following information may help to resolve the situation:
==> local: 
==> local: The following packages have unmet dependencies:
==> local:  php5-memcached : Depends: php5-common (>= 5.2.0) but it is not going to be installed
==> local:                   Depends: php5-common (< 6.0.0) but it is not going to be installed
==> local:                   Depends: phpapi-20121212
==> local: E: Unable to correct problems, you have held broken packages.
==> local: Notice: /Stage[main]/Puphpet_php/Service[php5.6-fpm]: Dependency Package[PhpModule_memcache] has failures: true
==> local: Notice: /Stage[main]/Puphpet_php/Service[php5.6-fpm]: Dependency Package[PhpModule_php5-memcached] has failures: true
==> local: Warning: /Stage[main]/Puphpet_php/Service[php5.6-fpm]: Skipping because of failed dependencies
==> local: Info: /Stage[main]/Puphpet_php/Service[php5.6-fpm]: Unscheduling all events on Service[php5.6-fpm]
```

Here is my `config.yaml` file:

```
vagrantfile:
    target: local
    vm:
        box: puphpet/ubuntu1404-x64
        box_url: puphpet/ubuntu1404-x64
        hostname: awesome.local
        memory: '512'
        cpus: '1'
        chosen_provider: virtualbox
        box_version: ''
        network:
            private_network: 192.168.56.101
            forwarded_port:
                vflnp_w44av6db6qjt:
                    host: '7329'
                    guest: '22'
                vflnp_nufh9jhk32c0:
                    host: '8080'
                    guest: '80'
        post_up_message: ''
        provider:
            virtualbox:
                modifyvm:
                    natdnshostresolver1: false
                showgui: '0'
            vmware:
                numvcpus: 1
            parallels:
                use_linked_clone: '0'
                check_guest_tools: '0'
                update_guest_tools: '0'
        provision:
            puppet:
                manifests_path: puphpet/puppet/manifests
                module_path: puphpet/puppet/modules
                options:
                    - '--verbose'
                    - '--hiera_config /vagrant/puphpet/puppet/hiera.yaml'
        synced_folder:
            vflsf_kb9zrcpivchb:
                source: ./
                target: /var/www
                sync_type: default
                smb:
                    smb_host: ''
                    smb_username: ''
                    smb_password: ''
                    mount_options:
                        dir_mode: '0775'
                        file_mode: '0664'
                rsync:
                    args:
                        - '--verbose'
                        - '--archive'
                        - '-z'
                    exclude:
                        - .vagrant/
                        - .git/
                    auto: 'true'
                owner: www-data
                group: www-data
        usable_port_range:
            start: 10200
            stop: 10500
    ssh:
        host: null
        port: null
        private_key_path: null
        username: vagrant
        guest_port: null
        keep_alive: true
        forward_agent: false
        forward_x11: false
        shell: 'bash -l'
        insert_key: false
    vagrant:
        host: detect
    proxy:
        http: ''
        https: ''
        ftp: ''
        no_proxy: ''
        enabled: ''
server:
    install: '1'
    packages:
        - ant
users_groups:
    install: '1'
    groups: {  }
    users: {  }
locale:
    install: '1'
    settings:
        default_locale: en_GB.UTF-8
        locales:
            - en_GB.UTF-8
            - en_US.UTF-8
        timezone: UTC
firewall:
    install: '1'
    rules: {  }
cron:
    install: '1'
    jobs: {  }
nginx:
    install: '0'
    settings:
        default_vhost: 1
        proxy_buffer_size: 128k
        proxy_buffers: '4 256k'
        proxy_connect_timeout: 600s
        proxy_send_timeout: 600s
        proxy_read_timeout: 600s
        names_hash_bucket_size: 128
    upstreams: {  }
    vhosts:
        nxv_q77yxhdfkhji:
            server_name: awesome.dev
            server_aliases:
                - www.awesome.dev
            www_root: /var/www/awesome
            listen_port: '80'
            index_files:
                - index.html
                - index.htm
                - index.php
            client_max_body_size: 1m
            ssl: '0'
            ssl_cert: ''
            ssl_key: ''
            ssl_port: '443'
            ssl_protocols: ''
            ssl_ciphers: ''
            rewrite_to_https: '1'
            spdy: '1'
            locations:
                nxvl_0t7s9viincws:
                    location: /
                    autoindex: 'off'
                    internal: 'false'
                    try_files:
                        - $uri
                        - $uri/
                        - /index.php$is_args$args
                    fastcgi: ''
                    fastcgi_index: ''
                    fastcgi_split_path: ''
                    proxy: ''
                    proxy_redirect: ''
                nxvl_hplkcvgyrm6p:
                    location: '~ \.php$'
                    autoindex: 'off'
                    internal: 'false'
                    try_files:
                        - $uri
                        - $uri/
                        - /index.php$is_args$args
                    fastcgi: '127.0.0.1:9000'
                    fastcgi_index: index.php
                    fastcgi_split_path: '^(.+\.php)(/.*)$'
                    fast_cgi_params_extra:
                        - 'SCRIPT_FILENAME $request_filename'
                        - 'APP_ENV dev'
                    set:
                        - '$path_info $fastcgi_path_info'
                    proxy: ''
                    proxy_redirect: ''
    proxies: {  }
apache:
    install: '1'
    settings:
        version: 2.4
        user: www-data
        group: www-data
        default_vhost: true
        manage_user: false
        manage_group: false
        sendfile: 0
    modules:
        - proxy_fcgi
        - rewrite
    vhosts:
        av_6fgcmv6n5ssk:
            servername: awesome.local
            docroot: /var/www/www
            port: '80'
            setenv:
                - 'APP_ENV dev'
            setenvif:
                - 'Authorization "(.*)" HTTP_AUTHORIZATION=$1'
            custom_fragment: ''
            ssl: '0'
            ssl_cert: ''
            ssl_key: ''
            ssl_chain: ''
            ssl_certs_dir: ''
            ssl_protocol: ''
            ssl_cipher: ''
            directories:
                avd_7mojet41ed3p:
                    path: /var/www/www
                    options:
                        - Indexes
                        - FollowSymlinks
                        - MultiViews
                    allow_override:
                        - All
                    require:
                        - 'all granted'
                    custom_fragment: ''
                    files_match:
                        avdfm_4khh3rrxybco:
                            path: \.php$
                            sethandler: 'proxy:fcgi://127.0.0.1:9000'
                            custom_fragment: ''
                            provider: filesmatch
                    provider: directory
php:
    install: '1'
    settings:
        version: '56'
    modules:
        php:
            - cli
            - intl
            - common
            - memcache
            - memcached
        pear: {  }
        pecl: {  }
    ini:
        display_errors: 'On'
        error_reporting: '-1'
        session.save_path: /var/lib/php/session
        date.timezone: UTC
    fpm_ini:
        error_log: /var/log/php-fpm.log
    fpm_pools:
        phpfp_5lueehx9jvdb:
            ini:
                prefix: www
                listen: '127.0.0.1:9000'
                security.limit_extensions: .php
                user: www-user
                group: www-data
    composer: '1'
    composer_home: ''
xdebug:
    install: '1'
    settings:
        xdebug.default_enable: '1'
        xdebug.remote_autostart: '0'
        xdebug.remote_connect_back: '1'
        xdebug.remote_enable: '1'
        xdebug.remote_handler: dbgp
        xdebug.remote_port: '9000'
blackfire:
    install: '0'
    settings:
        server_id: ''
        server_token: ''
        agent:
            http_proxy: ''
            https_proxy: ''
            log_file: stderr
            log_level: '1'
        php:
            agent_timeout: '0.25'
            log_file: ''
            log_level: '1'
xhprof:
    install: '0'
wpcli:
    install: '0'
    version: v0.21.1
drush:
    install: '0'
    version: 6.3.0
ruby:
    install: '1'
    versions:
        rv_jxyc5893x4vn:
            default: '1'
            version: 2.1.2
python:
    install: '1'
    packages: {  }
    versions: {  }
nodejs:
    install: '0'
    settings:
        version: '5'
    npm_packages: {  }
hhvm:
    install: '0'
    nightly: 0
    composer: '1'
    composer_home: ''
    settings: {  }
    server_ini:
        hhvm.server.host: 127.0.0.1
        hhvm.server.port: '9000'
        hhvm.log.use_log_file: '1'
        hhvm.log.file: /var/log/hhvm/error.log
    php_ini:
        display_errors: 'On'
        error_reporting: '-1'
        date.timezone: UTC
mariadb:
    install: '0'
    settings:
        version: '10.1'
        root_password: '123'
        override_options: {  }
    adminer: 0
    users:
        mariadbnu_drw4oxsycii8:
            name: dbuser
            password: '123'
    databases:
        mariadbnd_px1sbra5wh59:
            name: dbname
            sql: ''
    grants:
        mariadbng_nz88gtv7fnog:
            user: dbuser
            table: '*.*'
            privileges:
                - ALL
mysql:
    install: '1'
    settings:
        version: '5.5'
        root_password: 123
        override_options: {  }
    adminer: 0
    users: {  }
    databases: {  }
    grants: {  }
postgresql:
    install: '0'
    settings:
        global:
            encoding: UTF8
            version: '9.4'
        server:
            postgres_password: '123'
    databases: {  }
    users: {  }
    grants: {  }
    adminer: 0
mongodb:
    install: '0'
    settings:
        bind_ip: 127.0.0.1
        port: '27017'
    globals:
        version: 2.6.0
    databases: {  }
redis:
    install: '0'
    settings:
        port: '6379'
sqlite:
    install: '0'
    adminer: 0
    databases: {  }
mailhog:
    install: '0'
    settings:
        smtp_ip: 0.0.0.0
        smtp_port: 1025
        http_ip: 0.0.0.0
        http_port: '8025'
        path: /usr/local/bin/mailhog
beanstalkd:
    install: '0'
    settings:
        listenaddress: 0.0.0.0
        listenport: '11300'
        maxjobsize: '65535'
        maxconnections: '1024'
        binlogdir: /var/lib/beanstalkd/binlog
        binlogfsync: null
        binlogsize: '10485760'
    beanstalk_console: 0
rabbitmq:
    install: '0'
    settings:
        port: '5672'
    users: {  }
    vhosts: {  }
    plugins: {  }
elastic_search:
    install: '0'
    settings:
        version: 2.1.0
        java_install: true
    instances:
        esi_r5vztqsxo8z0:
            name: es-01
solr:
    install: '0'
    settings:
        version: 4.10.2
        port: '8984'


```

**Comment by elliotwms** [comment]
Could this be related to #2158 ?

**Comment by john-sparwasser** [comment]
I'm getting this same error when I try to install Redis. My same config.yaml worked a couple of weeks ago and now is broken.

**Comment by john-sparwasser** [comment]
Update! So I've been provisioning a bunch of vagrants this morning testing different configs looking for a workaround. If I put php5-common in the system packages, I'm able to get a successful provision.

**Comment by Ninestein** [comment]
+1 driving me nuts

**Comment by weilex** [comment]
+1

**Comment by Ninestein** [comment]
I solved this by rolling back to 5.5, but still required some apt-get updates to get it working. Not ideal, but is php 5.5.32 for now.

**Comment by jtreminio** [comment]
Fixed as of latest release.
